app/graphql/loaders/fancy_loader/type_generator.rb

```diff
@@ -0,0 +1,41 @@
+##
+# Generate parameter types for sorting
+class Loaders::FancyLoader::TypeGenerator
+  class SortDirection < GraphQL::Schema::Enum
+    graphql_name 'SortDirectionEnum'
+
+    value :ASCENDING, value: :asc
+    value :DESCENDING, value: :desc
+  end
+
+  def initialize(loader, name: loader.model.name)
+    @loader = loader
+    @name = name
+  end
+
+  def sorts_enum
+    sorts = @loader.sorts
+    name = "#{@name}SortEnum"
```

Comment:
can we not just memoize these or does that not play nicely when creating dynamic classes?